crates/ruff_python_parser/src/token.rs

```diff
@@ -16,6 +16,9 @@ pub enum Tok {
     /// Token value for a name, commonly known as an identifier.
     Name {
         /// The name value.
+        ///
+        /// Unicode names are NFKC-normalized by the lexer,
+        /// matching [the behaviour of Python's lexer](https://docs.python.org/3/reference/lexical_analysis.html#identifiers)
         name: Box<str>,
```

Comment:
I was eventually hoping to remove all owned data from the lexer tokens (e.g., prior to this change, we could've conceivably removed this field altogether; if we remove more similar fields from other tokens, we can eventually reduce the size of the `Tok` enum, which could be great for performance). This now precludes us from doing so. But I don't have enough context on the future design of the lexer-parser to know if it matters.

Reply:
I think we can still accomplish this and `Name` isn't the only token that must return the parsed value (e.g. FStrings etc).

What I have in mind is to:

* Replace `Tok` with `TokenKind` (that holds no data)
* Add a `take_value()` method on `Lexer` that "takes" the current value (`enum` over `Name`, `Int` etc.). 

This design also fits better into our new parser that already does exactly this internally (except that it "takes" the value from `Tok`). The advantage of this is that we only pay the overhead of reading or writting a value if it is a value token (and we're interested in the value).